network/p2p/connManager.go

```diff
@@ -87,3 +98,47 @@ func (c ConnManager) logConnectionUpdate(n network.Network, con network.Conn, lo
 		Int("total_connections", len(n.Conns())).
 		Msg(logMsg)
 }
+
+// ProtectPeer increments the stream setup count for the peer.ID
+func (c *ConnManager) ProtectPeer(id peer.ID) {
```

Comment:
when a stream is being created, we make it as `protected` by calling `ProtectPeer`